drivers/scale_host.go

```diff
@@ -222,13 +272,23 @@ func (s *ScaleHostDriver) Execute(conf interface{}, apiClient *client.RancherCli
 			}
 
 			name := basePrefix + currNameSuffix
-			hostRaw["name"] = ""
-			hostRaw["hostname"] = name
+			if config.HostTemplateID != "" {
+				hst.Name = ""
+				hst.Hostname = name
+				log.Infof("Creating host with hostname: %s", name)
 
-			log.Infof("Creating host with hostname: %s", name)
-			code, err := createHost(hostRaw, hostCreateURL, httpClient, cattleConfig.CattleAccessKey, cattleConfig.CattleSecretKey)
-			if err != nil {
-				return code, err
+				_, err := apiClient.Host.Create(hst)
+				if err != nil {
+					return http.StatusInternalServerError, nil
```

Comment:
You can't just swallow the error. Log it here and return an error message like "Error creating host"